SimulationAgent/SimulationManager.cs

```diff
@@ -228,6 +234,26 @@ public void PrintStats()
                 });
         }
 
+        public async Task SaveStatisticsAsync()
+        {
+            try
+            {
+                var simulationModel = new SimulationStatisticsModel
+                {
+                    TotalMessagesSent = this.deviceTelemetryActors.Where(a => a.Key.StartsWith(simulation.Id)).Sum(a => a.Value.TotalMessagesCount),
+                    FailedMessagesCount = this.deviceTelemetryActors.Where(a => a.Key.StartsWith(simulation.Id)).Sum(a => a.Value.FailedMessagesCount),
+                    FailedDeviceConnectionsCount = this.deviceConnectionActors.Where(a => a.Key.StartsWith(simulation.Id)).Sum(a => a.Value.FailedDeviceConnectionsCount),
+                    FailedDeviceTwinUpdatesCount = this.devicePropertiesActors.Where(a => a.Key.StartsWith(simulation.Id)).Sum(a => a.Value.FailedTwinUpdatesCount),
+                };
```

Comment:
what happens when the list of actors is empty? e.g. imagine there is a node simulating a single device, and the device is deleted. Will this return 0s or throw an exception?

Reply:
added null check for actors